### PR TITLE
[Slack] Handle properly "missing_scope" errors

### PR DESCRIPTION
**Pull request by nesk**
Getting user informations requires the `identity.*` scopes, however we might want to not add them to the scope list for various reasons.

Instead of throwing an exception on this error, we return an empty user.

This is a non-breaking changement.

### CODE REVIEW

**Comment by faustbrian** [comment]
Please use the `array_get` method instead of `Arr::get` to keep it consistent with all other providers.

**Comment by nesk** [comment]
Done, I've rebased the commit.

File: src/Slack/Provider.php [hunk 1]
```diff
@@ -106,9 +106,24 @@ protected function getTokenUrl()
...
+                'https://slack.com/api/users.identity?token='.$token
+            );
+        } catch (RequestException $exception) {
+            // Getting user informations requires the "identity.*" scopes, however we might want to not add them to the
+            // scope list for various reasons. Instead of throwing an exception on this error, we return an empty user.
+
+            if ($exception->hasResponse()) {
+                $data = json_decode($exception->getResponse()->getBody(), true);
+
+                if (Arr::get($data, 'error') === 'missing_scope') {
```
Comment (sevastyanovio) [inline src/Slack/Provider.php]:
```
FatalThrowableError in Provider.php line 120:
Class 'SocialiteProviders\Slack\Arr' not found
```

Comment (sevastyanovio) [inline src/Slack/Provider.php]: should be `if (\Arr::get($data, 'error') === 'missing_scope') {`

Comment (sevastyanovio) [inline src/Slack/Provider.php]:
```
FatalThrowableError in Provider.php line 120:
Class 'SocialiteProviders\Slack\Arr' not found
```

**Comment by nesk** [comment]
Oops, I missed that one. I've rebased the commit once again to get rid of all the calls to `Arr::get`. This time it should be alright.

**Pull request merged**